ephios/core/forms/users.py

```diff
@@ -480,3 +482,38 @@ def clean_url(self):
         if not url.endswith("/"):
             url += "/"
         return url
+
+
+class IdentityProviderForm(ModelForm):
+    client_secret = forms.CharField(
+        widget=PasswordInput(attrs={"placeholder": "********"}),
+        required=False,
+        label=_("Client secret"),
+        help_text=_("Leave empty to keep the current secret."),
+    )
+
+    class Meta:
+        model = IdentityProvider
+        fields = [
+            "label",
+            "client_id",
+            "client_secret",
+            "scopes",
+            "default_groups",
+            "group_claim",
+            "create_missing_groups",
+            "authorization_endpoint",
+            "token_endpoint",
+            "userinfo_endpoint",
+            "end_session_endpoint",
+            "jwks_uri",
+        ]
+        widgets = {
+            "default_groups": Select2MultipleWidget,
+        }
+
+    def clean_client_secret(self):
+        client_secret = self.cleaned_data["client_secret"]
+        if client_secret == "":
+            return self.instance.client_secret
+        return client_secret
```

Comment:
I didn't see this form being used in the create view. I think it would be good to use the same form both in update and create view, although this clean function is not suited for the create view.

Reply:
that was a deliberate decision :D i thought it would be easier to just leave it that way than custoimzing this form so that it also fits the create